### ISSUE TITLE
Data Grid: Problems with printing on mobile devices

### ISSUE THREAD

**Issue by pkuenemann** [mui/mui-x#7713]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://mui.com/x/

Problems with Chrome on mobile devices, Steps:
1. Open the website on an iPad Pro or Samsung Tablet S7 with Chrome
2. Scroll down to the Data Grid example an press Export/Print
3. The web page content will be printed instead of the list

Problems with Safari on iPad Pro, Steps:
1. Open the website on an iPad Pro or Samsung Tablet S7 with Chrome
2. Scroll down to the Data Grid example an press Export/Print
3. Last line on page 2 and first line on page 3 are truncated (not readable)

### Current behavior 😯

There are several issues with the printing functionality on mobile devices. The above described ones are just a few. Some of the issues (like truncated lines or extra blank pages) are also happening on Laptops with Windows 11 with Google Chrome (latest versions) when you use the page scaling option.

### Expected behavior 🤔

No matter wich device, browser or option, the data grid should be printed as expected: The grid itself should be printed with no lines missing and no extra blank pages.

### Context 🔦

We are offering a studio management solution for tattoo and piercing studios and there a lot of our customers are working with tablets. Since some of our data grids are affecting their accounting, there must not be missing any data on the printouts.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by MBilalShafi** [comment]
@pkuenemann Thanks for reporting, I was also able to reproduce this one on iPhone SE with Chrome and Firefox browsers. Though in Safari it looked fine.

**Comment by MBilalShafi** [comment]
> try with different printers and border settings and if available (depending on iOS Version I guess) the scaling factor (like 90%)

Yes I can see issues with Safari too, I've added it to our backlog, should be picked as soon as some bandwidth is available.

**Comment by pkuenemann** [comment]
...cool, thanx!

**Comment by pkuenemann** [comment]
It's a while ago I reported this bug. Any action taken so far? Would be nice to get this solved... Greets, Peter.

**Comment by pkuenemann** [comment]
Hey Guys, it is a bit disappointing that there is no update on a reported misbehaviour for one year now. You said " should be fixed as soon as some bandwidth is available". So no available bandwidth within one year?

**Comment by pkuenemann** [comment]
Dear MUI team, it is now more than two years ago that I reported this issue. meanwhile the printing problem spreaded out to windows browsers as well (Chrome on Windows 11). And you guy still lean back and do nothing out the printing issue of your product? Serious? Greets, Peter.
